Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php

```diff
@@ -33,14 +34,12 @@ public function updateIndexedContent(ContentEvent $event)
     {
         $content = $event->getContent();
 
-        if (!$content->getStatus()->isPublished()) {
-            return;
-        }
-
         $lastPublishedContent = $this->contentRepository->findLastPublishedVersion($content->getContentId(), $content->getLanguage());
 
-        if ($content->getVersion() >= $lastPublishedContent->getVersion()) {
-            $this->contentIndexor->index($content);
+        if ($lastPublishedContent instanceof ContentInterface) {
```

Comment:
I think too there could be an optimization here. There is three cases :
1/ the event gives a published content => index
2/ the event gives a content changing from status publish to something else => delete
3/ all other cases => do nothing